src/common/tse.c

```diff
@@ -854,7 +868,7 @@ tse_task_complete(tse_task_t *task, int ret)
 		if (done)
 			tse_task_complete_locked(dtp, dsp);
 	} else {
-		tse_task_decref_locked(dtp);
+		tse_task_decref_free_locked(task);
```

Comment:
here the task should at least have one refcount right? because the last one refcount suppose to be dropped in below tse_sched_process_complete().

Reply:
im not sure about that. i did reproduce the leak in the test that i added.
but you might be right that it hits the one in tse_task_post_process instead of this one.
that being said, i don't think it's an issue as this is checking the ref count before freeing.